pkg/uptimerobot.go

```diff
@@ -112,6 +112,34 @@ func (a AlertContact) String() string {
 	return render(alertContactTemplate, a)
 }
 
+// Status corresponds to the reported status of a monitor from the uptimerobot API
+type Status int
```

Comment:
Am I understanding this correctly? You want me to make a `map[int]string` for the statuses?

const enum

```go
monitors := getMonitors()

for m := range monitors {
    switch m.Status.(type) {
	case Up:
              // Do something
	case down:
             // This is a bug, but the compiler catches it
	default:
                 fmt.Println("Ignored")
	}
    }
}
```

With a map

```
monitors := getMonitors()

for m := range monitors {
   status, ok := MonitorStatuses[m.Status]
   if !ok {
      // Unknown status
   }

  if status == "Up" {
    // Do something
  }

  if status == "down" {
    // this is a bug, no compile time check
  }
}
```

In the second case also, MonitorStatus is also a package scoped `var`, [which I understood is not usually a great idea](https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables).

Reply:
Not quite; I’m saying MonitorTypes is a map of int to string, so let’s be consistent with that for now, rather than creating a new type definition for Status. Then if we decide to make custom types for all the various int fields, we can do that throughout the code, in one commit.

Reply:
Our defined constants will just be ints, rather than Statuses, and we'll still be comparing against the constants, not the strings they map to.

Reply:
So here's what I've done on the refactoring branch:

```go
// TypeHTTP represents an HTTP monitor.
const TypeHTTP = 1

// TypeKeyword represents a keyword monitor.
const TypeKeyword = 2

// TypePing represents a ping monitor.
const TypePing = 3

// TypePort represents a port monitor.
const TypePort = 4

// MonitorTypes maps an integer monitor type to the name of the monitor type.
var MonitorTypes = map[int]string{
	TypeHTTP:    "HTTP",
	TypeKeyword: "keyword",
	TypePing:    "ping",
	TypePort:    "port",
}
```

This seems like a nice compromise to me. What do you think about doing something similar for Status?